### PR TITLE
Show All Communities of filter list

### PR DESCRIPTION
**Pull request by abdullai-t**
#### Changes
- [x] Added a custom filter select component to all tables that is able to pull more communities from the backend.

#### Demo
https://github.com/massenergize/frontend-admin/assets/42780120/782a06cd-b2e3-4c78-9ff4-299a2ebdd838

### CODE REVIEW

File: app/utils/useObserver.js
Comment (frimpongopoku) [inline app/utils/useObserver.js:1]: @abdullai-t  we need comments here. Whats going on in "userObserver"? Whats the logic? And when you've explained the logic, how does anyone make use of "useObserver" anywhere else right away?

File: app/utils/components/AsyncSelect/AsyncSelect.js
Comment (frimpongopoku) [inline app/utils/components/AsyncSelect/AsyncSelect.js:1]: Again, same here, whats the concept. How does "AnySelect" work?  comments...? (You shouldnt be the only one that can easily make changes to this component 💀 )

**Review by frimpongopoku — CHANGES_REQUESTED** [review]
also @abdullai-t  conflicts

**Review by BradHN1 — APPROVED** [review]
LGTM

**Pull request merged**